File: tests/test_mastodon.rb
require 'test/unit'

require_relative 'utils'

# Tests for some routes that are used by Mastodon to find
# my social.alexwlchan.net server
class TestMastodon < Test::Unit::TestCase
  def test_host_meta
    resp = get_url('https://alexwlchan.net/.well-known/host-meta')

    assert_equal resp.code, '301'
    assert_equal resp['location'], 'https://social.alexwlchan.net/.well-known/host-meta'
  end

  def test_webfinger
    resp = get_url('https://alexwlchan.net/.well-known/webfinger')

    assert_equal resp.code, '301'
    assert_equal resp['location'], 'https://social.alexwlchan.net/.well-known/webfinger'
  end

  def test_nodeinfo
    resp = get_url('https://alexwlchan.net/.well-known/nodeinfo')

    assert_equal resp.code, '301'
    assert_equal resp['location'], 'https://social.alexwlchan.net/.well-known/nodeinfo'
  end
end

File: uptime_tests/test_mastodon.py
"""
Test routes used to find my Mastodon server at social.alexwlchan.net.
"""

import httpx
import pytest


@pytest.mark.parametrize(
    "path",
    ["/.well-known/host-meta", "/.well-known/webfinger", "/.well-known/nodeinfo"],
)
def test_mastodon_redirect(path: str) -> None:
    """
    Paths used by Mastodon instances are redirected to social.alexwlchan.net.
    """
    resp = httpx.get(f"https://alexwlchan.net{path}")

    assert resp.status_code == 301
    assert resp.headers["location"] == f"https://social.alexwlchan.net{path}"
